### PR TITLE
OAuth endpoints need @Controller annotations.

### PR DESCRIPTION
**Pull request by zawn**
The default display custom pac4j client.

fix remove refresh_token errors, modifications may have worked, of course, the best way is not to judge by type.

OAuth has been unable to work, add some missing annotations.

### CODE REVIEW

File: cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AccessTokenController.java [hunk 1]
```diff
@@ -41,13 +43,14 @@
  */
 @RefreshScope
 @Component("accessTokenController")
+@Controller
 public class OAuth20AccessTokenController extends BaseOAuthWrapperController {
```
Comment (mmoayyed) [inline cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AccessTokenController.java]: Change this to @Controller("accessTokenController")

File: cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AuthorizeController.java [hunk 2]
```diff
@@ -37,6 +39,7 @@
  */
 @RefreshScope
 @Component("authorizeController")
+@Controller
 public class OAuth20AuthorizeController extends BaseOAuthWrapperController {
```
Comment (mmoayyed) [inline cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AuthorizeController.java]: Change this to @Controller("authorizeController")

File: cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20CallbackAuthorizeController.java [hunk 3]
```diff
@@ -23,6 +24,7 @@
  */
 @RefreshScope
 @Component("callbackAuthorizeController")
+@Controller
 public class OAuth20CallbackAuthorizeController extends AbstractController {
```
Comment (mmoayyed) [inline cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20CallbackAuthorizeController.java]: Same here

File: cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20ProfileController.java [hunk 4]
```diff
@@ -28,7 +29,8 @@
  */
 @RefreshScope
 @Component("profileController")
-public class OAuth20ProfileController extends BaseOAuthWrapperController {
+@Controller
+public class    OAuth20ProfileController extends BaseOAuthWrapperController {
```
Comment (mmoayyed) [inline cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20ProfileController.java]: And here, fix the space.

**Comment by mmoayyed** [comment]
And fix the build failures please.

**Comment by zawn** [comment]
It has been modified

**Comment by leleuj** [comment]
I missed the change, but the context wrapper (https://github.com/Jasig/cas/blob/master/cas-server-support-oauth/src/main/java/org/jasig/cas/OAuthApplicationContextWrapper.java) does not declare the endpoints any more, so `@Component` must now be `@Controller`. @mmoayyed : when did the endpoints stop to be declared in the context wrapper?

**Comment by mmoayyed** [comment]
I believe that came about with the boot changes to make the code base more annotation friendly. Oidc makes similar changes. Controllers are expected to be annotated and should declare their own handlers via RequestMapping annotations. The patch here really addresses a regression with the boot changes.

File: cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20CallbackAuthorizeController.java (hunk 3 above)
Comment (mmoayyed) [inline cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20CallbackAuthorizeController.java]: Looks like this one is not updated yet.

File: cas-server-webapp-config/src/main/java/org/jasig/cas/config/CasProtocolViewsConfiguration.java [hunk 5]
```diff
@@ -139,7 +139,7 @@ public View cas3ServiceFailureView() {
     @RefreshScope
     @Bean(name = "oauthConfirmView")
     public View oauthConfirmView() {
-        return new CasProtocolView("protocol/oauth/confirm.jsp", this.applicationContext, this.springTemplateEngine);
```
Comment (zawn) [inline cas-server-webapp-config/src/main/java/org/jasig/cas/config/CasProtocolViewsConfiguration.java]: Forget to change the reference

**Comment by zawn** [comment]
Although references have changed, but I'm here garbled @mmoayyed ,response not set to UTF-8 encoding

File: cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20CallbackAuthorizeController.java (hunk 3 above)
Comment (zawn) [inline cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20CallbackAuthorizeController.java]: Sorry, I forgot this endpoint

**Comment by mmoayyed** [comment]
Excellent. Thanks very much :+1:

**Comment by zawn** [comment]
@mmoayyed Can you help solve garbled protocol/oauth/confirm page

thank you

**Comment by mmoayyed** [comment]
How?

**Comment by zawn** [comment]
image

Currently OAuth confirm page garbled.

**Comment by mmoayyed** [comment]
check the encoding on the messages file for your locale.

**Comment by zawn** [comment]
@mmoayyed Other pages are good, this is the only page garbled.

image

**Comment by zawn** [comment]
image

**Comment by mmoayyed** [comment]
File a separate issue please. We should force the view resolver to use UTF-8 when possible.

File: cas-server-webapp/src/main/resources/templates/fragments/loginsidebar.html [hunk 6]
```diff
@@ -22,8 +22,10 @@ <h3 th:utext="#{screen.welcome.label.loginwith}" />
                         <li th:case="'cas'" th:class="${'fa fa-sign-in'}">
                             <a th:href="${entry.redirectUrl}" th:text="${entry.name}" />
                         </li>
+                        <li th:case="*" th:class="${'fa fa-sign-in'}">
```
Comment (mmoayyed) [inline cas-server-webapp/src/main/resources/templates/fragments/loginsidebar.html]: What does this do? If the type is unrecognized, show it anyway? and use a generic icon?

Comment (zawn) [inline cas-server-webapp/src/main/resources/templates/fragments/loginsidebar.html]: image


Yes, I configured a client through the spring, of course I want to be displayed on the screen.
Even a little ugly does not matter, at least I know my configuration is correct.

Comment (mmoayyed) [inline cas-server-webapp/src/main/resources/templates/fragments/loginsidebar.html]: Gotcha. Looks fine. We may add a few more default ones.

**Pull request merged**